discovery/gossiper.go

```diff
@@ -928,16 +929,14 @@ func (d *AuthenticatedGossiper) networkHandler() {
 			// Next we check if we have any premature announcements
 			// for this height, if so, then we process them once
 			// more as normal announcements.
-			d.Lock()
 			numPremature := len(d.prematureAnnouncements[uint32(newBlock.Height)])
-			d.Unlock()
+			d.pAnnMtx.Unlock()
```

Comment:
Looks like we're missing a `d.pAnnMtx.Unlock()` on the other side of this line. Same for the usage below.

Reply:
Hm, not sure what happened here... Fixed!